beancount_import/matching.py

```diff
@@ -318,24 +435,30 @@ def _get_matches(
             self, account: str, date: datetime.date, amount: Amount,
             is_date_exact: bool) -> DatabaseValues:
         matches = dict() # type: DatabaseValues
-        for adjusted_date in self.get_fuzzy_date_range(date):
-            cur_matches = self._postings.get(adjusted_date)
-            if cur_matches is not None:
-                for key, (entry, mp) in cur_matches.items():
-                    posting = mp.posting
-                    # Verify that the account is compatible.
-                    if not are_accounts_mergeable(account, posting.account):
-                        continue
+        dc = (date, amount.currency)
+        upper = amount.number + self.fuzzy_match_amount
+        lower = amount.number - self.fuzzy_match_amount
+
+        cur_matches = self.get_date_currency_postings(dc)
+        if cur_matches is not None:
+            lower_bound = bisect.bisect_left(cur_matches, (lower, tuple(), None, None))
+            upper_bound = bisect.bisect_right(cur_matches, (upper, (sys.maxsize,), None, None), lo=lower_bound)
+            for sp in cur_matches[lower_bound-1 if lower_bound > 0 else 0:upper_bound]:
```

Comment:
@xentac It seems like this `lower_bound-1` adjustment will bring in an extra element that falls outside the bounds, causing https://github.com/jbms/beancount-import/pull/206#issuecomment-1789816679. Do you remember why it was necessary?